compat/test/browser/Children.test.js

```diff
@@ -99,6 +99,16 @@ describe('Children', () => {
 			expect(serializeHtml(scratch)).to.equal('<div></div>');
 		});
 
+		it('should propagate the this context', () => {
```

Comment:
This is a great start, but I think the commit isn't complete. The test defines a few functions but never calls them anywhere. This will make the test always pass, even against current master and without the changes in this PR.

```suggestion
		it('should propagate this context', () => {
			const context = {};
			const spy = sinon.spy(child => child); // noop
			const Foo = ({ children }) => {
				return React.Children.map(children, spy, context);
			};
			render(<Foo>foo</Foo>, scratch);

			expect(spy.thisValues[0]).to.equal(context);
		});
```